### ISSUE TITLE
"WSL Service" service changes to disabled within 60 seconds of a reboot

### ISSUE THREAD

**Issue by SidJnsn66** [microsoft/WSL#11628]
### Windows Version

Microsoft Windows [Version 10.0.19045.4412]

### WSL Version

2.2.4.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

 5.15.153.1-2

### Distro Version

Ubuntu-22.04

### Other Software

None

### Repro Steps

The WSL Service is set to Manual.  Reboot the system.  Open the Services window and see that it is still set to manual.  Refresh the window repeatedly.  Within 60 seconds, it will changed to Disabled.

### Expected Behavior

The service should remain as manual, and should run when an Ubuntu terminal window is opened.

### Actual Behavior

It changes to disabled, and nothing WSL works until it is manually changed back to manual or auto.
Note the following:
Sc.exe config WSLService start= demand makes no difference, nor does changing the start value in the registry directly
LxssManager is auto and running
Avast Antivirus has been completely removed.
Added an exception to Win Defender, and finally turned off Defender A/V. No change.
Docker has been completely removed.
Disabling the Security Health Service makes no difference.
Disabling my VPN service made no difference.
Only started happening within the last couple of weeks
Removing all distros and reinstalling wsl made no difference

### Diagnostic Logs

[WslLogs-2024-05-27_21-35-17.zip](https://github.com/microsoft/WSL/files/15463350/WslLogs-2024-05-27_21-35-17.zip)

**Comment by SidJnsn66** [comment]
I'm 99% sure I caught the problem with this log.  IDK what a wsl trace is; however, I'm wondering if there would be one given that, on power-up, WSL Service is not running until there is a demand.  If I quickly open a terminal, WSL Service will start running and there is no problem.  The problem occurs if there is no demand within about a minute, at which time WSL Service changes to Disabled.  So WSL never runs and cannot be made to run without changing the service back to manual or auto.

If I'm missing something, please let me know.  Thx

**Comment by OneBlue** [comment]
Thank you for reporting this @SidJnsn66. 

This is most likely caused by a third party program (there's no logic to ever disable wslservice), so this is most likely an antivirus / security solution. 

Could you share the list of all running process & services right after boot ? That should help scope down where this is coming from.

**Comment by SidJnsn66** [comment]
That makes sense.  Thanks for hanging with me to get closer to a solution.  Here is the output of tasklist right after reboot and before WSL Service switches to Disabled.  
As mentioned above, I have uninstalled Avast AV and, at one time or another, have  turned off Defender, disabled Secure Health Service, turned of PIA (vpn client) and CCleaner, all with no effect.

[list.txt](https://github.com/microsoft/WSL/files/15476843/list.txt)

**Comment by OneBlue** [comment]
@SidJnsn66: I see a few processes that could potentially explain this service configuration change: 

- CCleanerPerformanceOptimi     
- MpDefenderCoreService
- MsMpEng
- CCleaner64

Could you try disabling those and see if that solves the issue ?

**Comment by SidJnsn66** [comment]
Seems the problem is solved before I got your last message.  I decided to clean up the registry using CCleaner in case traces of Avast were left and problematic.  Apparently something in that did the trick.  WSL service was Automatic - Delayed Start when I next rebooted, and never returned to Disabled.  WSL runs as expected.
FWIW, the four services you mentioned are still running.
Thank you for your help!!

**Comment by irowbin** [comment]
@OneBlue I had the exact issue and it seems this `ccleaner` performance optimizer who puts services into sleep and if i put sleep to `Microsoft Visual Studio Tool For Application 2019`, the `WSL` service becomes disabled and when undo the sleep, it back to `automatic`  

Thanks for the hints above so I could figure it out.

**Comment by EmTheSKills** [comment]
If I can provide more help for those who are still using Avast anti-virus and their `WSL` keeps getting disabled, I simply disabled the `Avast Cleanup` service, and now everything works as expected! My `WSL` stopped getting disabled every 60 seconds, and everything is running smoothly.

Afterward, you can always enable the `Avast Cleanup` service again if you want to tweak the performance of your computer, and disable it when you're done!

Here's the service to disable:
![Image](https://github.com/user-attachments/assets/0c86bcee-abb1-4780-b90c-15d2b818cd12)

**Comment by boaza** [comment]
# `WslService` keeps getting disabled at boot (`Wsl/0x80070422`) — and it's not WSL

**TL;DR:** If `WslService` keeps reverting to **Disabled** on every boot (`Wsl/0x80070422`), stop debugging WSL. Check what *else* auto-starts and hooks the network stack. In my case it was **Twingate**.

The error names WSL, so every instinct (and every LLM I asked) sends you *inward*: reinstall WSL, repair the optional feature, check the hypervisor, blame the latest cumulative update, suspect Docker. I burned hours on all of that. None of it was the cause.

## Symptom

- `wsl` → `The service cannot be started, either because it is disabled... Error code: Wsl/0x80070422`
- `sc.exe qc WslService` shows `START_TYPE : 4 DISABLED`
- You set it back to demand/auto, it works, then after the *next reboot* it's `DISABLED` again
- Reinstalling WSL, toggling the feature, `DISM /RestoreHealth`, `sfc /scannow` — none of it sticks

## How to find the real culprit

Audit writes to the service's registry key so you catch *what* disables it:

```powershell
auditpol /set /subcategory:"Registry" /success:enable
```

Then in `regedit` on `HKLM\SYSTEM\CurrentControlSet\Services\WslService` → **Permissions → Advanced → Auditing** → add `Everyone`, Type `Success`, advanced permissions → check **Set Value** only.

Reboot, then:

```powershell
Get-WinEvent -FilterHashtable @{LogName='Security'; Id=4657} -MaxEvents 30 |
  Where-Object { $_.Message -match 'WslService' } | Format-List TimeCreated, Message
```

This logs the process writing `Start` to `4`. In my case it was `services.exe` (the Service Control Manager) — meaning Windows itself was applying the disabled state at boot, with no WSL-side failure reason. That ruled out WSL being broken and pointed at *something instructing the service config*.

> **Honest caveat:** the audit caught `services.exe` (the SCM), **not** Twingate writing the value directly. So the precise mechanism — how Twingate's boot activity causes the SCM to disable `WslService` — is inferred from the elimination test, not proven from the log. But setting Twingate to manual is what reproducibly fixes it, so the boot-time conflict is the trigger.

## The actual cause

Twingate (a zero-trust / VPN client) was installed with its **service set to auto-start**. It launches early in boot, hooks the network stack, and collides with WSL2's virtual-network init — leaving `WslService` disabled. WSL2 depends on the host network / ICS plumbing, and the VPN client was winning the race at boot.

## The fix

Set the Twingate service to **manual** start so it isn't racing WSL at boot:

```powershell
Stop-Service Twingate.Service -Force
Set-Service Twingate.Service -StartupType Manual
sc.exe config WslService start= auto
```

After this, WSL comes up cleanly on every boot. Started manually *after* boot, Twingate and WSL coexist fine — I can run WSL, Docker, and Twingate simultaneously. It was specifically the **boot-time auto-start race** that broke it.

## Takeaway for the next person (or LLM)

The error blames WSL, but if it keeps re-disabling at boot, the cause is almost certainly an *external* service that auto-starts and touches networking — VPN / ZTNA clients especially. **Audit the registry key, find the writer, set that other service to manual. Don't reinstall WSL for the tenth time.**

---

## Optional: PowerShell helper functions

Drop these in your profile (`notepad $PROFILE`). They require an **elevated** PowerShell.

```powershell
function Assert-Admin {
    if (-not ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)) {
        Write-Host "This needs an elevated PowerShell (Run as Administrator)." -ForegroundColor Yellow
        return $false
    }
    return $true
}

function twingate-on {
    if (-not (Assert-Admin)) { return }
    Start-Service Twingate.Service
    Start-Process "C:\Program Files\Twingate\Twingate.exe" -ErrorAction SilentlyContinue
    Write-Host "Twingate service + app started." -ForegroundColor Green
}

function twingate-off {
    if (-not (Assert-Admin)) { return }
    Get-Process | Where-Object ProcessName -match 'twingate' | Stop-Process -Force -ErrorAction SilentlyContinue
    Stop-Service Twingate.Service -Force
    sc.exe config WslService start= auto | Out-Null
    Start-Service WslService -ErrorAction SilentlyContinue
    Write-Host "Twingate stopped, WSL restored." -ForegroundColor Green
}

function wsl-fix {
    if (-not (Assert-Admin)) { return }
    sc.exe config WslService start= auto | Out-Null
    Start-Service WslService -ErrorAction SilentlyContinue
    wsl --status
}
```
